From f14e1cacc55394d6e4c5073db42951b95b23bf44 Mon Sep 17 00:00:00 2001
From: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
Date: Mon, 4 May 2026 07:20:55 +0000
Subject: [PATCH] EDAC/i10nm: fix refcount leak in pci_get_dev_wrapper()

jira VULN-168502
cve CVE-2022-50645
commit-author Yang Yingliang <yangyingliang@huawei.com>
commit 9c8921555907f4d723f01ed2d859b66f2d14f08e

As the comment of pci_get_domain_bus_and_slot() says, it returns
a PCI device with refcount incremented, so it doesn't need to
call an extra pci_dev_get() in pci_get_dev_wrapper(), and the PCI
device needs to be put in the error path.

Fixes: d4dc89d069aa ("EDAC, i10nm: Add a driver for Intel 10nm server processors")
	Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
	Reviewed-by: Qiuxu Zhuo <qiuxu.zhuo@intel.com>
	Signed-off-by: Tony Luck <tony.luck@intel.com>
Link: https://lore.kernel.org/r/20221128065512.3572550-1-yangyingliang@huawei.com
(cherry picked from commit 9c8921555907f4d723f01ed2d859b66f2d14f08e)
	Signed-off-by: CIQ Kernel Automation <ciq_kernel_automation@ciq.com>
---
 drivers/edac/i10nm_base.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/edac/i10nm_base.c b/drivers/edac/i10nm_base.c
index 765d7c0b992c..345ba6a93c27 100644
--- a/drivers/edac/i10nm_base.c
+++ b/drivers/edac/i10nm_base.c
@@ -198,11 +198,10 @@ static struct pci_dev *pci_get_dev_wrapper(int dom, unsigned int bus,
 	if (unlikely(pci_enable_device(pdev) < 0)) {
 		edac_dbg(2, "Failed to enable device %02x:%02x.%x\n",
 			 bus, dev, fun);
+		pci_dev_put(pdev);
 		return NULL;
 	}
 
-	pci_dev_get(pdev);
-
 	return pdev;
 }
 
-- 
2.47.3

